gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMethodDescriptor.java

```diff
@@ -131,14 +131,86 @@ public ResponseT read(JsonReader in) throws IOException {
   }
 
   ResponseT parseResponse(Reader input) {
-    return responseMarshaller().fromJson(input, responseType());
+    return getResponseMarshaller().fromJson(input, getResponseType());
   }
 
   void writeRequest(Appendable output, RequestT request) {
-    this.requestMarshaller().toJson(request, output);
+    this.getRequestMarshaller().toJson(request, output);
   }
 
   void writeRequestBody(RequestT apiMessage, Appendable output) {
-    httpRequestBuilder().writeRequestBody(apiMessage, requestMarshaller(), output);
+    getHttpRequestBuilder().writeRequestBody(apiMessage, getRequestMarshaller(), output);
+  }
+
+  public static <RequestT, ResponseT> Builder<RequestT, ResponseT> newBuilder() {
+    return new Builder<>();
+  }
+
+  public final Builder<RequestT, ResponseT> toBuilder() {
+    return new Builder<>();
+  }
+
+  public static class Builder<RequestT, ResponseT> {
```

Comment:
You can use AutoValue.Builder to cut out a lot of boilerplate here

Reply:
How do you make an AutoValue class with a AutoValue.Builder when the getter methods have different signatures from the setters? e.g. only the Builder deals with RequestT, and only the main class deals with java.lang.reflect.Type.

I tried making a separate class called ApiMethodDescriptorFactory and that did allow me to use symmetrical setters and getters in AutoValue/AutoValue.Builde. That resulted in a net increase of 30 LOC and just seemed like even more boilerplate, so I scrapped it.

Reply:
Ok sounds fine. I have also had several circumstances where I had to ditch AutoValue.Builder because of certain constraints.